roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py

```
@@ -0,0 +1,93 @@
+                self._sampling_frequency = self._video.attrs["fr"]
+                self._start_time = self._video.attrs["start_time"]
+                self.metadata = self._video.attrs["meta_data"]
+            else:
+                raise Exception(f"{file_path} does not contain the 'mov' dataset")
+
+        self._num_channels, self._num_frames, self._size_x, self._size_y = get_video_shape(self._video)
+
+        if len(self._video.shape) == 3:
+            # check if this converts to np.ndarray
+            self._video = self._video[np.newaxis, :]
+
+        if self._channel_names is not None:
+            assert len(self._channel_names) == self._num_channels, "'channel_names' length is different than number " \
+                                                                   "of channels"
+        else:
+            self._channel_names = [f'channel_{ch}' for ch in range(self._num_channels)]
+
+        self._kwargs = {'file_path': str(Path(file_path).absolute()), 'mov_field': mov_field,
+                        'sampling_frequency': sampling_frequency, 'channel_names': channel_names}
+
+    def get_frame(self, frame_idx, channel=0):
+        assert frame_idx < self.get_num_frames()
+        return self._video[channel, frame_idx]
+
+    def get_frames(self, frame_idxs, channel=0):
+        frame_idxs = np.array(frame_idxs)
+        assert np.all(frame_idxs < self.get_num_frames())
+        sorted_frame_idxs, sorting_inverse = np.sort(frame_idxs, return_inverse=True)
+        return self._video[channel, sorted_frame_idxs][:, sorting_inverse]
+
+    # TODO make decorator to check and correct inputs
+    def get_video(self, start_frame=None, end_frame=None, channel=0):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_frames()
+        end_frame = min(end_frame, self.get_num_frames())
+
+        video = self._video[channel, start_frame: end_frame]
```

You can just make `start_frame` and `end_frame` `None` here. You don't need to re-assign. Also, I think making `end_frame` too high should throw an error

---

@bendichter isn't that exactly how it is? `start_frame` and `end_frame` are `None` by default and set to 0 and `end_frame` if nor given.

What should be a good value to raise and error/warning if `end_frame-start_frame` is to high? 10 000?

---

`np_array[1, None:None]` is the same as `np_array[1, 0:-1]` You can just pass the default `None`s right into the index and it'll work as you want.